packages/amplify-category-api/src/commands/api/generate-schema.ts

```diff
@@ -20,47 +21,72 @@ const subcommand = 'generate-schema';
 
 export const name = subcommand;
 
-export const run = async (context: $TSContext) => {
+export const run = async (context: $TSContext): Promise<void> => {
   printer.warn(PREVIEW_BANNER);
-  const apiName = getAppSyncAPIName();
-  const apiResourceDir = getAPIResourceDir(apiName);
+  const sqlSchema = context.parameters?.options?.['sql-schema'];
+  const engineType = context.parameters?.options?.['engine-type'];
+  const out = context.parameters?.options?.out;
+  // unauthenticated flow
+  if (sqlSchema || engineType || out) {
+    if (!(sqlSchema && engineType && out)) {
+      if (!sqlSchema) {
+        printer.error('A SQL schema must be provided with --sql-schema');
+      }
+      if (!engineType) {
+        printer.error('An engine type must be provided with --engine-type');
+      }
+      if (!out) {
+        printer.error('An output path must be provided with --out');
+      }
+      return;
```

Comment:
Do we have a pattern for abnormal exits (e.g., exiting with a non-zero status) to indicate errors? Can we throw an error rather than just returning?

Reply:
This is what is used for other failures. See the migrate command: https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-category-api/src/commands/api/migrate.ts

This probably should be changed though because it give a 0 exit code on the cli. Throwing an error will still call `printer.error`, so I'll change to that.